testing/async/exclusive_scan/large_indices.cu

```diff
@@ -0,0 +1,239 @@
+#include <thrust/detail/config.h>
+
+#if THRUST_CPP_DIALECT >= 2014
+
+#include <async/test_policy_overloads.h>
+
+#include <async/exclusive_scan/mixin.h>
+
+#include <thrust/device_free.h>
+#include <thrust/device_malloc.h>
+#include <thrust/device_ptr.h>
+#include <thrust/optional.h>
+
+#include <thrust/iterator/detail/device_system_tag.h>
+#include <thrust/iterator/detail/iterator_facade_category.h>
+
+#include <cstdint>
+
+// This test is an adaptation of TestInclusiveScanWithBigIndices from scan.cu.
+
+namespace
+{
+
+// Fake iterator that asserts
+// (a) it is written with a sequence and
+// (b) a defined maximum value is written at some point
+struct assert_sequence_iterator
```

Comment:
I'd prefer to have a single implementation of this for both algorithms if possible to reduce code duplication.

Reply:
These iterator implementations are unique enough that we need different ones for the different scans.

I may try to abstract this later as we expand the new-style tests to more algorithms, we'll see how that goes.